catlearn/graph_utils.py

```diff
@@ -69,6 +69,27 @@ def op(self) -> DirectedGraph[NodeType]:
         """
         return self.reverse(copy=False)
 
+    @property
+    def dual(self) -> DirectedGraph[NodeType]:
```

Comment:
I don't like the name, we should find an other one (it's clearly not a dual of the graph; although relations are dualized)

Maybe don't make it a property and rename it dualize_relations ?